lib/apiservers/engine/backends/archive.go

```diff
@@ -447,7 +442,7 @@ func (wm *ArchiveStreamWriterMap) WriterForAsset(proxy proxy.VicArchiveProxy, ci
 			deviceID = aw.mountPoint.Name
 			store = constants.VolumeStoreName
 		}
-		rawWriter, err := proxy.ArchiveImportWriter(wm.op, store, deviceID, aw.filterSpec)
+		rawWriter, err := proxy.ArchiveImportWriter(wm.op, store, deviceID, aw.filterSpec, wm.errchan)
```

Comment:
We do catch the `io.Copy` error in `unpack()`. All the errors are returned but we weren't waiting for them before. The error channel is used to block personality from returning until all writes are finished.